## store infromation needed for all protocols
locations:
  bedroom:
    x: -1.08
    y: -1.4811692237854004
    yaw: -1.5708   # -90 degrees

  living_room:
    x: 1.0967974662780762
    y: 1.9037660360336304
    yaw: 0.0       # facing forward
  home:
    x: 2.6955928802490234 #0.85 #
    y: 3.3216073513031006 #1.014 #
    yaw: -1.5708   # -90 degrees

  dining:
    x: 5.066422939300537
    y: 3.4881248474121094
    yaw: -1.5708   # -90 degrees



## need to be integerated to have robot interface get its info 
person_location_state: "person_location"
robot_location_state: "robot_location"

# available type: ## todo set it in a way to verifiy htat yaml doesnt include 
## undefined types: text, audio, video, question_answer 
protocols:
  XReminderProtocol:
    medicine_am_2:
      low_level:
        reminder_1: "/home/hello-robot/smarthome_ws/src/smart_home_robot/smart_home_pytree/audio_resources/food_reminder_dain.mp3"
        type_1: audio
        wait_1: 300 ## 5 minutes
        reminder_2:  "please take your night medicine"
        type_2: text
        number_of_protocols: 2
      high_level:
        priority: 2
        reset_pattern:
          type: periodic
          minutes: 60
        requirements:
          time: {from: "10:00", to: "11:00"}
          permissible_locations: [living_room]

## add states with /
  CoffeeProtocol:
    coffee:
      low_level:
        first_text: "please clean the dishes"
      high_level:
        priority: 3
        reset_pattern:
          type: periodic
          hours: 1
          minutes: 30
        requirements: 
          event: 
            - {'state': 'coffee', 'value': True}
          time: {'from': '09:00', 'to': '13:00'}

  MoveAwayProtocol:
    move_away:
      low_level:
        away_location: dining # move_away has two loc, home or away where away is the location defined here and has to be part of the locations in the yaml
        position_state_key: position # key to indicate whether to move away or home
        state_key: move_away
      high_level:
        reset_pattern:
          type: instant 
        priority: 1
        requirements: 
          event: 
            - {'state': 'move_away', 'value': True}  

  ExerciseProtocol:
    exercise:
      low_level:
        exercise_yaml_file: "/home/hello-robot/smarthome_ws/src/smart_home_robot/smart_home_pytree/config/exercise_routine_test.yaml" 
        video_dir_path: "file:///storage/emulated/0/Download/Exercise_Videos" ## path on the tablet 
        start_state_key: "start_exercise"
        stop_state_key: "stop_exercise"
        get_confirmation: "Do you want to exercise now?" ## if empty then dont ask to indictae whther the protocol should start after the person asks
      high_level:
        reset_pattern:
          type: instant ## if triggered should play directly
        priority: 3
        requirements: 
          event: 
            - {'state': 'start_exercise', 'value': True}
          time: {'from': '15:00', 'to': '17:00'}

  ChargeRobot:
    charge_robot: 
      high_level:
        priority: 100
        requirements:
          event: 
              - {'state': 'charging', 'value': False}
## TO TEST new orchestrator, start at time 10:30 during the waiting i will trigger coffee true